### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by DLPerf**
issue: #13 
Thank you for replying. If there are many this kind of bugs, program will reduce obviously.
Check the PR, please. Looking forward to your confirmation.

### CODE REVIEW

**Comment by lreiher** [comment]
Can you quantify the impact that this change had on your system?

**Comment by DLPerf** [comment]
> Can you quantify the impact that this change had on your system?

It can be proved by the running time of the program.  I have verified what I said. The amount of reduction is determined by the code. I didn't test your code, but `tf.zeros(image.shape[0:2], dtype=tf.int32)` created in the loop did cause program redundancy.

**Comment by lreiher** [comment]
Sure, just wanted to ask whether you conducted two similar runs before/after the change and measured the reduction in elapsed time. Like is it a 1%, 5%, 10%, ... performance improvement?

**Comment by bastilam** [comment]
Thanks for your pull request!

I checked it and it truly leads to a substantial speedup.

Unfortunately, the code changes break the intended function of `one_hot_encode_image_op`. The original code makes it possible to combine multiple colors in an image into one class. This would not be possible anymore with the update.

Apart from that, it does not seem very useful to call the `bitwise_or` function in your adjusted line
`class_map = tf.bitwise.bitwise_or(c_map, tf.cast(tf.reduce_all(tf.equal(image, color), axis=-1), tf.int32))`
since in your update, `c_map` is always a tensor full of zeros, so it cannot have any influence on the result.

Next, the variable name `c_map` is not descriptive enough in my opinion.

Feel free to open a new pull request if you find another way to speed up the code while keeping the intended functionality. Thanks!
